internal/application/workload.go

```diff
@@ -106,8 +106,17 @@ func (b AppConfigurationBindList) ToNames() []string {
 func (a *Workload) Deployment(ctx context.Context) (*appsv1.Deployment, error) {
 	var err error
 	if a.deployment == nil {
-		a.deployment, err = a.cluster.Kubectl.AppsV1().
-			Deployments(a.app.Namespace).Get(ctx, a.app.Name, metav1.GetOptions{})
+		depList, err := a.cluster.Kubectl.AppsV1().
+			Deployments(a.app.Namespace).List(ctx, metav1.ListOptions{
+			LabelSelector: fmt.Sprintf("app.kubernetes.io/component=application,app.kubernetes.io/name=%s,app.kubernetes.io/part-of=%s", a.app.Name, a.app.Namespace),
+		})
+		if err != nil {
+			return nil, err
+		}
+		if len(depList.Items) < 1 {
```

Comment:
Should we check for `> 1` also ? I.e. ambiguous result, multiple matches.

Reply:
There are only 2 consumers of this method. One of the 2 (line 184 below in this file) ignores all errors except "not found". I will add the check for the other consumer though.

Reply:
done below